slog/slog.go

```diff
@@ -95,11 +98,32 @@ func Configure(cfg Config) error {
 		th.ReplaceAttr = func(groups []string, a slog.Attr) slog.Attr {
 			// Customize the name of some fields to match Google Cloud expectations
 			// More: https://cloud.google.com/logging/docs/agent/logging/configuration#process-payload
-			if a.Key == slog.LevelKey {
-				a.Key = "severity"
+			if len(groups) == 0 {
+				if a.Key == slog.LevelKey {
+					a.Key = "severity"
+				}
+				if a.Key == slog.MessageKey {
+					a.Key = "message"
+				}
 			}
-			if a.Key == slog.MessageKey {
-				a.Key = "message"
+
+			// Customize the http request fields
+			// More: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
+			if len(groups) > 0 && groups[0] == "httpRequest" {
```

Comment:
We may have multiple groups no ? you need to iterate on all groups to find "httpRequest" no ? (maybe that is why you didn't find the http_request ?). Even if on isolated tests we only have one group or this one is the first..Im not sure this will always be true depending on how users of the library use grouping/manipulate the loggers.

Reply:
Not really, but I also thought about this.

The `groups` means open groups... So, when the fields I'm interested in exists, the only open group available will be `httpRequest`, if I have another group inside of `httpRequest`, then `len(groups) == 2`. 

It's not all groups that exists!

Reply:
Ah OK, the groups is the whole path in the hierarchy of groups, so you are looking for the root group "httpRequest". I find this grouping thing kinda messy TBH...for what I checked playing around with the lib seems like it is only useful when you use the TextHandler..it is a way to group thins that the text handler understands and uses to serialize the log entries. But for JSON Handler it is redundant... a map/struct already does grouping perfectly fine. Unless you want to create a logger with a group and then pass the logger to some other function that then won't know it is logging inside a group...but I don't see a clear use case for that yet...I usually just add new fields to provide context..not groups...hmm